homeassistant/components/nobo_hub/strings.json

```diff
@@ -0,0 +1,66 @@
+{
+  "config": {
+    "step": {
+      "user": {
+        "description": "Start by entering the complete serial number (12 digits) of the Nobø Ecohub you want to add. If you have more than one hub, and they have the same serial prefix, you must also specify the IP address. If any devices were automatically found on your network, they are shown below.\n\nIf you cannot see your device or experience any issues, try specifying the device IP address.\n\n{devices}",
+        "data": {
+          "serial": "Serial number",
+          "ip_address": "[%key:common::config_flow::data::ip%]"
+        }
+      }
+    },
+    "error": {
+      "cannot_connect": "[%key:common::config_flow::error::cannot_connect%]",
+      "device_not_found": "Device not found",
+      "invalid_serial": "Invalid serial number",
+      "invalid_ip": "Invalid IP address",
+      "unknown": "[%key:common::config_flow::error::unknown%]"
+    },
+    "abort": {
+      "already_configured": "[%key:common::config_flow::abort::already_configured_device%]"
+    }
+  },
+  "options": {
+    "step": {
+      "init": {
+        "data": {
+          "override_type": "Override type",
+          "command_off": "Week profile with 'off' setting",
+          "command_on_zone_1": "Normal week profile for zone 1",
```

Comment:
High level question: Could it work to force the same profile name be used in every zone? Showing 20 different zone names seems not ideal

Reply:
Good question, as this way to configure the integration really bothered me. Before implementing 32 zones here I asked the Norwegian community about their various setups. The answer is that is depends on the user's configuration. A week profile describes for when the heaters in a zone should be off, comfort, and eco mode, controlled by the hub. Some (e.g. I) have made different week profiles per zone, while others let HA control this, and use the same week profile for `normal` for all zones. Also, the limit of how many zones the hub supports is undocumented, but if you want to control each heater individually, 32 sounds like a reasonable limitation. (The limitation is really only in the config flow - this would be much prettier if it was possible to provide a dynamic length list in the config flow.)

Allowing different "normal" program per zone is also what's currently supported in the HACS version (using `configuration.yaml`), so removing that would make it harder for some users to convert. For more details, see https://github.com/echoromeo/hanobo

Reply:
For some context - my current setup:
![image](https://user-images.githubusercontent.com/5984518/148702633-00eb3100-e2ba-4caa-a0db-1233381be532.png)

Reply:
It's hard to understand why we do it like this. Why aren't we using a select entity per zone to select the profile for that zone?

Is there some more limitation that we need to consider when deciding the design for this than:
1. Zones
2. Profiles

What is the matrix of options that we need to represent with our user interface?

Reply:
Configuring "normal" week profile per entity instead would be a nice solution, and clean up this mess.

Configuring the "off" week profile would be redundant to do on each entity, but it probably makes sense to do it the same place as the "normal" week profile is configured.

However, how do I build an option flow for a climate entity?

Reply:
Here's my personal list of zones and week profiles:

Zones:
- Living room
- Bedrooms
- Guest room
- Study
- Basement living room
- Bathroom

Week profiles:
- Default
- Study
- Basement living room
- Guest room
- Bath
- Bedrooms
- Living room
- Home
- Away
- Basement living room home
- Bedrooms home
- Off

Reply:
(Note the naming the normal week profile the same as the zone is just my personal naming convention.)

Reply:
As per Martins comment would probably look nicer if there is a select entity per zone where you select the profile instead of running it through a config/options flow

Reply:
@gjohansson-ST I agree, and I’m happy to implement that, but I need some pointers to where that should be plugged in and preferably some example where something similar is done. It doesn’t look like it’s possible to extend the entity configuration. The zones are represented as ClimateEntity instances.

Reply:
We should not support setting the HVAC mode in the climate entity. The profile feature of the zone doesn't match what we represent in Home Assistant climate entities with the HVAC mode.

The profile setting feature and select platform should not be included in this PR. That should be a follow up PR.

Look at existing integrations that implement a `select.py` platform. Here are the docs:

https://developers.home-assistant.io/docs/core/entity/select

Reply:
@MartinHjelmare The OptionFlow is not intended as a way to let the user switch between week profiles. It is a one-time setup to support HVAC mode to "off". This is based on the implementation in the HACS integration, where this is configured in `configuration.yaml`: https://github.com/echoromeo/hanobo/#there-is-no-override-in-nob%C3%B8-for-completely-off-but-there-is-a-workaround

Reply:
We shouldn't abuse setting the HVAC mode when the device doesn't support the same concept. It doesn't matter what the custom integration does. Custom integrations often don't implement things correctly according to our architecture design.

Reply:
Point taken. I'll remove this part of the configuration from the OptionFlow.